intelmq/bots/experts/tor_nodes/expert.py

```diff
@@ -43,5 +50,48 @@ def process(self):
         self.send_message(event)
         self.acknowledge_message()
 
+    @classmethod
+    def run(cls):
+        if len(sys.argv) > 1 and sys.argv[1] == "--update-database":
```

Comment:
As you maybe already imagined, I don't like that ;) The ideal solution would be a generic interface in the core which handles the argument parsing for this kind of things. However, I think that we can postpone this to future versions. It works reasonably well for now.

Reply:
Yes I can imagine. :-) This was just the most straight forward solution for a working proof of concept code.

Reply:
Should be improved now.